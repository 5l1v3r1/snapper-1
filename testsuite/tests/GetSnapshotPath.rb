# encoding: utf-8

module Yast
  class GetSnapshotPathClient < Client
    def main
      # testedfiles: Snapper.ycp

      Yast.import "Snapper"
      Yast.import "Testsuite"

      @READ = { "snapper" => { "path" => nil } }
      Testsuite.Test(lambda { Snapper.GetSnapshotPath(0) }, [@READ, {}, {}], 0)

      Ops.set(@READ, ["snapper", "path"], "/snapshots/0/snapshot")

      Testsuite.Test(lambda { Snapper.GetSnapshotPath(0) }, [@READ, {}, {}], 0)

      nil
    end
  end
end

Yast::GetSnapshotPathClient.new.main
