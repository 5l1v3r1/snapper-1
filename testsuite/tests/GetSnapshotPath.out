Read	.snapper.path $["num":0] nil
Log	Snapshot '0' was not found.
Return	
Read	.snapper.path $["num":0] "/snapshots/0/snapshot"
Return	/snapshots/0/snapshot
